tests/unit/test_charm.py

```diff
@@ -116,35 +117,65 @@ def test_remote_write_configuration(self, mock_push: MagicMock):
         )
 
         path, content = mock_push.call_args[0]
-        self.assertEqual(path, "/etc/agent/agent.yaml")
-        self.assertDictEqual(
-            yaml.load(content, Loader=yaml.FullLoader),
-            {
-                "integrations": {
-                    "agent": {
-                        "enabled": True,
-                        "relabel_configs": REWRITE_CONFIGS,
-                    },
-                    "prometheus_remote_write": [
-                        {"url": "http://1.1.1.2:9090/api/v1/write"},
-                        {"url": "http://1.1.1.1:9090/api/v1/write"},
-                    ],
-                },
-                "prometheus": {
-                    "configs": [
-                        {
-                            "name": "agent_scraper",
-                            "remote_write": [
-                                {"url": "http://1.1.1.2:9090/api/v1/write"},
-                                {"url": "http://1.1.1.1:9090/api/v1/write"},
-                            ],
-                            "scrape_configs": [],
-                        }
-                    ]
+        content = yaml.load(content, Loader=yaml.FullLoader)
+        expected_config: Dict[str, Any] = {
+            "integrations": {
+                "agent": {
+                    "enabled": True,
+                    "relabel_configs": REWRITE_CONFIGS,
                 },
-                "server": {"log_level": "info"},
+                "prometheus_remote_write": [
+                    {"url": "http://1.1.1.2:9090/api/v1/write"},
+                    {"url": "http://1.1.1.1:9090/api/v1/write"},
+                ],
             },
+            "prometheus": {
+                "configs": [
+                    {
+                        "name": "agent_scraper",
+                        "remote_write": [
+                            {"url": "http://1.1.1.2:9090/api/v1/write"},
+                            {"url": "http://1.1.1.1:9090/api/v1/write"},
+                        ],
+                        "scrape_configs": [],
+                    }
+                ]
+            },
+            "server": {"log_level": "info"},
+        }
+        self.assertEqual(path, "/etc/agent/agent.yaml")
+
+        # Since we are comparing two dictionaries that has lists of dictionaries inside,
+        # for instance:
+        #
+        # "remote_write": [
+        #     {"url": "http://1.1.1.2:9090/api/v1/write"},
+        #     {"url": "http://1.1.1.1:9090/api/v1/write"},
+        # ],
+        #
+        # and there is no guarantee that the lists will come in the same order every time
+        # we run the tests, we cannot use `self.assertDictEqual()` and we have
+        # to compare the dictionaries in parts.
```

Comment:
Maybe the value of this test would be increased if broken down into smaller pieces? Not sure.